src/vehicle_move.cpp

```diff
@@ -1009,6 +1011,121 @@ float get_collision_factor( const float delta_v )
     }
 }
 
+void vehicle::precalculate_vehicle_turning( int new_turn_dir, bool falling_only,
+        bool check_rail_direction,
+        const ter_bitflags ter_flag_to_check,
+        int &wheels_on_rail,
+        bool &all_wheels_on_one_axis,
+        int &turning_wheels_that_are_one_axis )
+{
+    tileray mdir; // The direction we're moving
+    mdir.init( new_turn_dir ); // calculate direction after turn
+    tripoint dp;
+    bool is_diagonal_movement = new_turn_dir % 90 == 45;
+
+    if( abs( velocity ) >= 20 && !falling_only ) {
+        mdir.advance( velocity < 0 ? -1 : 1 );
+        dp.x = mdir.dx();
+        dp.y = mdir.dy();
+    }
+
+    // number of wheels that will land on rail
+    wheels_on_rail = 0;
+
+    int yVal = INT_MAX;
+    // is all wheels of vehicle on one axis?(bike)
+    all_wheels_on_one_axis = true;
```

Comment:
calculate and cache this in vehicle::refresh()

Reply:
`yVal` can't be cached, it is used to calculate `turning_wheels_that_are_one_axis`